File: app/src/main/java/com/adda/addatest/viewmodel/UserViewModel.java
package com.adda.addatest.viewmodel;

import android.view.View;

import androidx.databinding.ObservableArrayMap;
import androidx.databinding.ObservableInt;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import com.adda.addatest.R;
import com.adda.addatest.adapter.UsersAdapter;
import com.adda.addatest.model.User;
import com.adda.addatest.model.Users;

import java.util.List;


public class UserViewModel extends ViewModel {

    private Users users;
    private UsersAdapter adapter;
    public MutableLiveData<User> selected;
    public ObservableArrayMap<String, String> images;
    public ObservableInt loading;
    public ObservableInt showEmpty;

    public void init() {
        users = new Users();
        selected = new MutableLiveData<>();
        adapter = new UsersAdapter(R.layout.list_item, this);
        images = new ObservableArrayMap<>();
        loading = new ObservableInt(View.GONE);
        showEmpty = new ObservableInt(View.GONE);
    }

    public void fetchList() {
        users.fetchList();
    }

    public MutableLiveData<List<User>> getUsers() {
        return users.getUsers();
    }

    public UsersAdapter getAdapter() {
        return adapter;
    }

    public void setUsersInAdapter(List<User> users) {
        this.adapter.setUsers(users);
        this.adapter.notifyDataSetChanged();
    }

    public MutableLiveData<User> getSelected() {
        return selected;
    }

    public void onItemClick(Integer index) {
        User db = getUserAt(index);
        selected.setValue(db);
    }

    public User getUserAt(Integer index) {
        if (users.getUsers().getValue() != null &&
                index != null &&
                users.getUsers().getValue().size() > index) {
            return users.getUsers().getValue().get(index);
        }
        return null;
    }

}

File: app/src/main/java/com/adda/addatest/viewmodel/UserViewModel.kt
package com.adda.addatest.viewmodel

import android.view.View
import androidx.databinding.ObservableArrayMap
import androidx.databinding.ObservableInt
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import com.adda.addatest.R
import com.adda.addatest.adapter.UsersAdapter
import com.adda.addatest.model.User
import com.adda.addatest.model.Users

class UserViewModel : ViewModel() {
    var users: Users? = null
    var adapter: UsersAdapter? = null
        private set
    lateinit var selected: MutableLiveData<User?>
    var images: ObservableArrayMap<String, String>? = null
    @JvmField
    var loading: ObservableInt? = null
    @JvmField
    var showEmpty: ObservableInt? = null
    fun init() {
        users = Users()
        selected = MutableLiveData()
        adapter = UsersAdapter(R.layout.list_item, this)
        images = ObservableArrayMap()
        loading = ObservableInt(View.GONE)
        showEmpty = ObservableInt(View.GONE)
    }

    fun fetchList() {
        users!!.fetchList()
    }

    fun getUsers(): MutableLiveData<List<User>> {
        return users!!.users
    }

    fun setUsersInAdapter(users: List<User>?) {
        adapter!!.setUsers(users)
        adapter!!.notifyDataSetChanged()
    }

    fun onItemClick(index: Int?) {
        val db = getUserAt(index)
        selected!!.value = db
    }

    fun getUserAt(index: Int?): User? {
        return if (users!!.users.value != null && index != null && users!!.users.value!!.size > index) {
            users!!.users.value!![index]
        } else null
    }
}